implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -49,22 +51,24 @@ public String getValue(String name) {
             return null;
         }
 
+        final Map<String, String> properties = getProperties();
+
         // exact match
-        String value = AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getenv(name));
```

Comment:
I see this approach results in better re-use, but not sure if it's a good idea to cache the environment properties?

Reply:
it's not cache, i'ts local copy at the time of the search. i think it's ok